src/OpenLoco/Windows/Options.cpp

```diff
@@ -1945,6 +1962,115 @@ namespace OpenLoco::Ui::Options
+
+            auto y = w->y + w->widgets[Widx::autosave_frequency].top + 1;
+            drawString_494B3F(*dpi, w->x + 10, y, Colour::black, StringIds::autosave_frequency, nullptr);
+
+            auto freq = Config::getNew().autosave_frequency;
+            string_id stringId;
+            switch (freq)
+            {
+                case 0:
+                    stringId = StringIds::autosave_never;
+                    break;
+                case 1:
+                    stringId = StringIds::autosave_every_month;
+                    break;
+                default:
+                    stringId = StringIds::autosave_every_x_months;
+                    break;
+            }
+            drawDropdownContent(w, dpi, Widx::autosave_frequency, stringId, freq);
+
+            y = w->y + w->widgets[Widx::autosave_amount].top + 1;
+            drawString_494B3F(*dpi, w->x + 10, y, Colour::black, StringIds::autosave_amount, nullptr);
+
+            auto scale = Config::getNew().autosave_amount;
+            drawDropdownContent(w, dpi, Widx::autosave_amount, StringIds::int_32, scale);
+        }
+
+        static void changeAutosaveAmount(window* w, int32_t delta)
+        {
+            auto& cfg = Config::getNew();
+            auto newValue = std::clamp(cfg.autosave_amount + delta, 1, 24);
+            if (cfg.autosave_amount != newValue)
+            {
+                cfg.autosave_amount = newValue;
+                Config::writeNewConfig();
+                w->invalidate();
+            }
+        }
+
+        static void changeAutosaveFrequency(window* w, int32_t value)
+        {
+            auto& cfg = Config::getNew();
+            if (cfg.autosave_frequency != value)
+            {
+                cfg.autosave_frequency = value;
+                Config::writeNewConfig();
+                w->invalidate();
+            }
+        }
+
+        static void showAutosaveFrequencyDropdown(window* w, widget_index wi)
+        {
+            auto dropdown = w->widgets[wi];
+            Dropdown::show(w->x + dropdown.left, w->y + dropdown.top, dropdown.width() - 4, dropdown.height(), w->colours[1], 5, 0x80);
+
+            // Add pre-defined entries
+            Dropdown::add(0, StringIds::dropdown_stringid, { StringIds::autosave_never });
+            Dropdown::add(1, StringIds::dropdown_stringid, { StringIds::autosave_every_month });
+            Dropdown::add(2, StringIds::dropdown_stringid, { StringIds::autosave_every_x_months, static_cast<uint16_t>(3) });
+            Dropdown::add(3, StringIds::dropdown_stringid, { StringIds::autosave_every_x_months, static_cast<uint16_t>(6) });
+            Dropdown::add(4, StringIds::dropdown_stringid, { StringIds::autosave_every_x_months, static_cast<uint16_t>(12) });
+
+            // Set current selection
+            auto freq = Config::getNew().autosave_frequency;
+            std::optional<size_t> selected;
+            switch (freq)
```

Comment:
What if the frequency is set to 2?

Reply:
Wouldn't it just use `StringIds::autosave_every_x_months`? Seems acceptable to me.

Reply:
Yes, the game only lets you set it to predefined values, but it can display any value if set in the config manually.